### PR TITLE
redux-form - Fix the broken type inheritance

### PR DESCRIPTION
**Pull request by mnemanja**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Basically, this pull request solves this request 
https://github.com/DefinitelyTyped/DefinitelyTyped/issues/27250

### CODE REVIEW

**Comment by mnemanja** [comment]
Can someone please assist with the failing tests?
Locally ran `npm test` produces the following:
```
Clean data
Clean logs
Clean output
# Typing Publish Report Summary
Started at Fri, 13 Jul 2018 11:18:03 GMT
Found 4651 typings folders in /Users/nemanja/www/DefinitelyTyped/types
Error: Directory /Users/nemanja/www/DefinitelyTyped/types/redux-form/.idea has unused file jsLibraryMappings.xml
    at /Users/nemanja/www/DefinitelyTyped/node_modules/types-publisher/src/lib/definition-parser.ts:385:11
    at Generator.next (<anonymous>)
    at fulfilled (/Users/nemanja/www/DefinitelyTyped/node_modules/types-publisher/bin/lib/definition-parser.js:4:58)
    at <anonymous>
Error: Parsing failed.
    at fail (/Users/nemanja/www/DefinitelyTyped/node_modules/types-publisher/src/util/util.ts:320:11)
    at ChildProcess.child.on (/Users/nemanja/www/DefinitelyTyped/node_modules/types-publisher/src/util/util.ts:308:6)
    at emitNone (events.js:106:13)
    at ChildProcess.emit (events.js:208:7)
    at finish (internal/child_process.js:757:14)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
npm ERR! Test failed.  See above for more details.
```

**Comment by mnemanja** [comment]
Hi guys,
@CarsonF , @aikoven , @LKay , @bancek , @tehbi4 , @huwmartin , @Reggino , @maddijoyce

Could you please check this pull request?

Thanks!
Nemanja

**Review by Reggino — APPROVED** [review]
lgtm :shipit:
